### PR TITLE
Fredrik/add support for notifications

### PR DESCRIPTION
**Pull request by fredrikolovsson**
Depends on https://github.com/DINA-Web/dina-semantic-ui/pull/8

### CODE REVIEW

File: src/coreModules/notifications/components/Flash.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const Flash = ({
+  descriptionKey,
+  headerKey,
+  removeNotification,
+  sequentialId,
+  i18n: { translate },
+  ttl,
+}) => {
+  if (ttl) {
+    setTimeout(() => removeNotification({ sequentialId }), ttl)
```
Comment (antonoberg) [inline src/coreModules/notifications/components/Flash.js]: We want do to this in component did mount and remove the timeout in component will unmount.

File: src/coreModules/form/components/Input.js [hunk 2]
```diff
@@ -57,14 +69,24 @@ const Field = ({
           {helpText && ' ('}
           {helpText && helpText}
           {helpText && ')'}
+          {helpText && helpNotification && ' '}
```
Comment (antonoberg) [inline src/coreModules/form/components/Input.js]: I think we should move this to a separate component

File: src/coreModules/notifications/reducer.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  addSequentialId,
+  insertInPriorityOrder,
+  removeBySequentialId,
+} from './utilities'
+
+const getActiveNotifications = createGetter(['activeNotifications'])
+const setActiveNotifications = createSetter(['activeNotifications'])
+
+const initState = {
+  activeNotifications: [],
```
Comment (antonoberg) [inline src/coreModules/notifications/reducer.js]: Refactor state to be an object and to related changes

File: src/coreModules/notifications/reducer.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        customRegisterKey: NOTIFICATIONS_REGISTER,
+        ignoreModuleNames: true,
+        property: NOTIFICATIONS,
+        state,
+      })
+      return nextState
+    }
+
+    case NOTIFICATIONS_CREATE_NOTIFICATION: {
+      const { type, ...specificationOverrides } = action.payload
```
Comment (antonoberg) [inline src/coreModules/notifications/reducer.js]: Move creation of the notification to the actionCreator instead and create the sequential id there instead. Might make sense to use a separate factory function here. This is a good place to set defaults

File: src/coreModules/user/notifications.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Flash } from 'coreModules/notifications/components'
+import { FIXED } from 'coreModules/notifications/constants'
+
+const LOG_IN_SUCCESS = {
+  component: Flash,
+  componentProps: {
+    headerKey: 'modules.user.welcomeBack',
```
Comment (antonoberg) [inline src/coreModules/user/notifications.js:7]: Add style: info to componentProps

File: src/coreModules/notifications/components/Flash.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Message } from 'semantic-ui-react'
+
+import { withI18n } from 'coreModules/i18n/higherOrderComponents'
+
+const propTypes = {
+  descriptionKey: PropTypes.string,
```
Comment (antonoberg) [inline src/coreModules/notifications/components/Flash.js]: Add componentOption style (or other name of the key) which is an enum of ['info', 'success', 'warn', 'error']

File: src/coreModules/user/notifications.js (hunk 5 above)
Comment (fredrikolovsson) [reply src/coreModules/user/notifications.js:7]: set info as default in Flash instead

**Pull request merged**